bigbite/themer
# [BBT-102] Custom CSS UI component

## Description



Completes [BBT-102](https://b5ecom.atlassian.net/browse/BBT-102) - This PR adds a reusable component that allows a user to add custom CSS on a per-block basis.

For now, I've just added a basic `textarea` component without any validation. As discussed in Slack, we can look to improve the code editing experience in a future ticket (after MVP?). I've also added a link to a WordPress blog post in the help section as the syntax is a little different to what you'd expect.

**NOTE:** I couldn't get this to work in the preview, only the frontend. I've experienced this with the filter PR (#28) too, so I'm thinking we might need to open up a new ticket to investigate these preview issues?

## Change Log



- Added 'Custom CSS' reusable UI component.

## Steps to test



- Add some custom CSS to one of your blocks in `theme.json`. E.g. 
```json
"blocks": {
    "core/paragraph": {
      "css": "font-size: 3rem;"
    },
}
```
- Check the `textarea` is pre-populated with your CSS when you load themer.
- Add some more CSS using the textarea.
- Check on the frontend to see if your styles have been applied. As previously stateed, I couldn't get this to work in the preview.
- Export theme.json and ensure it contains your updated styles.

## Screenshots/Videos



### Importing from `theme.json`

https://github.com/bigbite/themer/assets/45216035/2c5bd3a6-6f8e-477f-9697-93e418b5db34

### Changing values

https://github.com/bigbite/themer/assets/45216035/204e15d5-618d-4959-b224-d5357f62b072

### Exporting `theme.json`

https://github.com/bigbite/themer/assets/45216035/540ff22f-28aa-48a7-a0dd-0bf03a533662

## Checklist:





[BBT-102]: https://b5ecom.atlassian.net/browse/BBT-102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
